drivers/common/opensensors.c

```diff
@@ -0,0 +1,25 @@
+#include "opendefs.h"
+#include "opensensors.h"
+
+opensensors_vars_t opensensors_vars;
+
+void opensensors_init(void) {
+   memset(&opensensors_vars,0,sizeof(opensensors_vars_t));
+}
+
+void opensensors_register(uint8_t path1len,
+                      uint8_t* path1val,
+                      callbackRead_cbt callbackRead,
+                      callbackConvert_cbt callbackConvert) {
+   opensensors_vars.opensensors_resource[opensensors_vars.numSensors].path1len        = path1len;
+   opensensors_vars.opensensors_resource[opensensors_vars.numSensors].path1val        = path1val;
+   opensensors_vars.opensensors_resource[opensensors_vars.numSensors].callbackRead    = callbackRead;
+   opensensors_vars.opensensors_resource[opensensors_vars.numSensors].callbackConvert = callbackConvert;
+   opensensors_vars.numSensors++;
+}
+
+opensensors_vars_t* opensensors_read(void) {
```

Comment:
The new code includes a opensensors_getNumSensors (returning the number of sensors available) and opensensors_getResource, which takes an index as argument (the position of the sensor in the array) and returns a opensensor_resource structure (i.e., sensorType, callbackRead and callbackConvert). A pointer to this structure is registered as attribute in a csensors_resource.